.github/workflows/ci.yml

```diff
@@ -0,0 +1,43 @@
+name: ci
+
+# Default to 'contents: read', which grants actions to read commits.
+#
+# If any permission is set, any permission not included in the list is
+# implicitly set to "none".
+#
+# see https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - 'master'
```

Comment:
Note to self: we can probably rename the branch to `main` on this repository. We could already anticipate that happening and add both `master` and `main`, or I can already rename the branch; wdyt?